task/mgmt-gateway/src/update/sp.rs

```diff
@@ -131,8 +227,74 @@ impl SpUpdate {
         }
 
         // Copy fields of `current` so we can borrow it mutably.
+        let aux_flash_size = current.aux_flash_size;
         let sp_image_size = current.sp_image_size;
 
+        // Handle aux flash states.
+        if let Some(result) = current.update_state_with_result(|state| {
+            let auxflash_state = match state {
+                State::AuxFlash(s) => s,
+                // All other states are handled below.
+                _ => return (state, None),
+            };
+
+            // We are in an aux flash state - is this chunk aux flash data?
+            if *component != SpComponent::SP_AUX_FLASH {
+                return (
+                    State::AuxFlash(auxflash_state),
+                    Some(Err(ResponseError::InvalidUpdateChunk)),
+                );
+            }
+
+            // Are we in a state where we can accept data?
+            let accepting = match auxflash_state {
+                AuxFlashState::AcceptingData(a) => a,
+                AuxFlashState::FinishedErasingSlot(s) => {
+                    s.into_accepting_data()
+                }
+                AuxFlashState::ScanningForChck(_)
+                | AuxFlashState::ErasingSlot(_) => {
+                    return (
+                        State::AuxFlash(auxflash_state),
+                        Some(Err(ResponseError::UpdateNotPrepared)),
+                    );
+                }
+                AuxFlashState::Failed(err) => {
+                    return (
+                        State::AuxFlash(auxflash_state),
```

Comment:
Do we want to fall into `State::Failing` here, instead of the nested fail state?

Reply:
I don't think so, but it doesn't _really_ matter since the only way to recover from landing in a `Failed` state (of either variety) is to abort the update and restart it. `State::Failed(_)` holds an `UpdateError` but `AuxFlashState::Failed(_)` holds an `AuxFlashError`; to fall into `State::Failed(_)` we'd have to discard the aux flash error and pick an update error instead (probably `UpdateError::FlashError`, which is what I use in another place where we have to do this).